sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIO.java

```diff
@@ -155,17 +168,53 @@
  * FhirIO.Write.Result writeResult =
  *     output.apply("Execute FHIR Bundles", FhirIO.executeBundles(options.getExistingFhirStore()));
  *
+ * // Alternatively you could use import for high throughput to a new store.
+ * FhirIO.Write.Result writeResult =
+ *     output.apply("Import FHIR Resources", FhirIO.executeBundles(options.getNewFhirStore()));
+ * // [End Writing ]
+ *
  * PCollection<HealthcareIOError<String>> failedBundles = writeResult.getFailedInsertsWithErr();
  *
+ * // [Begin Writing to Dead Letter Queue]
  * failedBundles.apply("Write failed bundles to BigQuery",
  *     BigQueryIO
  *         .write()
  *         .to(option.getBQFhirExecuteBundlesDeadLetterTable())
  *         .withFormatFunction(new HealthcareIOErrorToTableRow()));
+ * // [End Writing to Dead Letter Queue]
+ *
+ * // Alternatively you may want to handle DeadLetter with conditional update
+ * // [Begin Reconciliation with Conditional Update]
+ * failedBundles
+ *     .apply("Reconcile with Conditional Update",
+ *         FhirIO.ConditionalUpdate(fhirStore)
+ *             .withFormatBodyFunction(HealthcareIOError<String>::getDataResource)
+ *             .withTypeFunction((HealthcareIOError<String> err) -> {
+ *               String body = err.getDataResource();
+ *               // TODO(user) insert logic to exctract type.
+ *               return params;
```

Comment:
I see that [type can be Patient or Observation?](https://cloud.google.com/healthcare/docs/reference/rest/v1beta1/projects.locations.datasets.fhirStores.fhir/create) - is this an exhaustive list? Does it make sense to have an ENUM for the types that we expect? It would prevent issues with random strings being returned?

Reply:
These are just examples. For R4 the exhaustive list is [here](https://www.hl7.org/fhir/valueset-resource-types.html).

I don't want to add an enum because I think different versions of FHIR may provide different resource types and I want FhirIO implementation be pretty version agnostic. Though we test against all versions, the code path is identical.

If we were to add an enum I wouldn't want to maintain it as part of beam. 

If we want to institute stronger typing and domain sanity checks, we could consider adding a dependency on the open source [HAPI library](https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-structures-r4/org/hl7/fhir/r4/model/package-summary.html) which is maintained by the HL7 community and is pretty popular.